src/components/UI/ModalWindow.jsx

```diff
@@ -0,0 +1,56 @@
+import { Button, Grid, Modal, styled, Typography } from '@mui/material/'
+import { useState } from 'react'
+
+export const ModalWindow = ({
+   modalTitle,
+   bodyContent,
+   headerContent,
+   footerContent,
+}) => {
+   const [open, setOpen] = useState(false)
```

Comment:
that state will be created when someone will call your modal component
Also button which will open the modal